docs/proposal-component-names.md

```diff
@@ -0,0 +1,62 @@
+# Proposal (Feb 2020)
+
+## Cylc 8 System Component Naming
+
+Cylc is now a complex multi-component system. It is important that we settle on
+consistent, unambiguous, and where possible standard, component names so that
+users and developers can communicate without confusion. (This was not easy even
+before Cylc 8!).
+
+## Proposed Final Names
+
+Firstly, we have already agreed to drop **suite** for **workflow** throughout
+("suite" is non-standard outside of the NWP community).
+
+Secondly, we should name the various system components as follows (changes marked with :boom:):
+
+| Component Name               | Code Repository   | Config File                                          | Description |
+| --                           |----               | ---                                                  |---          |
+|:boom: **Cylc Flow Instance** | cylc-flow         | :boom: `cylc-flow.conf` and `cylc-flow-global.conf`  | an instance of the cylc-flow program, to manage a single workflow | 
+|**Cylc Hub**                  | (Jupyter Hub)     | :boom: `cylc-hub.conf`                               | the single point of access that users log in to |
+|**Cylc UI Server**            | cylc-uiserver     | N/A? (or :boom: `cylc-uiserver.conf`)                | an instance of the cylc-uiserver program, that serves the Cylc Web UI and CLI |
+|:boom: **Cylc Web UI**        | :boom: cylc-webui | N/A?                                                 | an instance of the Cylc Web UI, served to your browser by the Cylc UI Server |
+
+
+## Explanatory Comments
+
+Cylc Flow Instance
+- "workflow server" is ambiguous - it can refer to the host server that the workflows run on
+- "workflow server program" is clearer, but too much of a mouthful.
+- "workflow daemon" is not strictly correct (`--no-detach`) and is very
+  Unix-specific (should we ever port to another OS)
+- "workflow service" - a cylc-flow instance really stretches the definition of a "service"
+- "cylc-flow instance" is strictly correct and unambiguous, reasonably
+  intuitive (a running workflow instance), and its function in the context of the wider system
+  can be described more verbosely if needed (see the Description in the table above)
+- :question: "Flow Instance" is an option
+  - (but "Cylc Flow Instance" is clearer without being too verbose)
+
+Cylc config files: `cylc-flow.conf`, `cylc-flow-global.conf`
```

Comment:
can we modify this so that a `cylc-flow-user.conf` and `cylc-flow-site.conf` are used to create the global config object? 

Also is the "flow" necessary (or even a good idea) in the global config names?

Reply:
> can we modify this so that a cylc-flow-user.conf and cylc-flow-site.conf are used to create the global config object?

I think that's only necessary if there are real differences between the user and site global config (i.e. are we proposing site global config items that users cannot override in their global config file? ... if not, making the `-user.conf` and `-site.conf` distinction seems a bit superfluous to me.

> Also is the "flow" necessary (or even a good idea) in the global config names?

 My reasoning here was that the global and workflow config files both configure the same program, i.e. the "cylc flow instance" aka WFS etc.  So the config file name contains the name of the program that it configures.  I guess it's not strictly necessary, but if it is not overly verbose or ungainly, then it is (IMO) nicely self-documenting in that respect.